### PR TITLE
feat/debug menu, fix: localization and fonts

### PR DESCRIPTION
**Pull request by ThinhKimVo**
Screen Shot 2023-02-14 at 16 04 18

### CODE REVIEW

**Review by baonguyenhsts — APPROVED** [review]

File: template/scripts/fonts.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const generate = () => {
   const string = `const fonts = {
   ${properties},
 }
-export default fonts
+export {fonts}
```
Comment (huydosgtech) [inline template/scripts/fonts.js:27]: why export {fonts} and not use default?

File: template/App.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from 'react'
 import {Provider} from 'react-redux'
-import {PersistGate} from 'redux-persist/integration/react'
 import {store, persistor} from './src/store/store'
+import {PersistGate} from 'redux-persist/lib/integration/react'
```
Comment (loc-nguyenthien) [inline template/App.js:4]: Why are we changing this? The example in the latest docs is still `import {PersistGate} from 'redux-persist/integration/react'`

File: template/src/components/DebugMenu.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+import {InfoMenu, InfoMenuRow, InfoMenuLink} from './InfoMenu'
+
+import {colors, fonts, metrics} from '../themes'
+
+import {localize} from '../locale/I18nConfig'
+import {useDispatch, useSelector} from 'react-redux'
+import {getApiUrl, getCodePushKey} from '../store/selectors'
+import {appActions} from '../store/reducers'
+
+const debugMenuSize = 50
```
Comment (loc-nguyenthien) [inline template/src/components/DebugMenu.js:28]: I think we can move this to file `constants`

File: template/src/components/DebugMenu.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+const AppInfoSection = () => {
+  const appName = useMemo(() => getApplicationName(), [])
+  const buildNumber = useMemo(() => getBuildNumber(), [])
+  const deviceId = useMemo(() => getDeviceId(), [])
```
Comment (loc-nguyenthien) [inline template/src/components/DebugMenu.js:33]: Same above, move to `constants`

File: template/src/components/DebugMenu.js [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    ],
+    [appName, buildNumber, deviceId],
+  )
+
+  return (
+    <View style={styles.section}>
+      <Text style={styles.h3}>{localize('debug.info')}</Text>
+      <View style={styles.content}>
+        {infos.map(({title, description}) => (
+          <InfoMenuRow style={styles.infoMenu} title={title} description={description} />
```
Comment (loc-nguyenthien) [inline template/src/components/DebugMenu.js:52]: Missing `key` prop

File: template/src/components/DebugMenu.js [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+  const currentApiUrl = useSelector(getApiUrl)
+
+  const [bottomSheetType, setBottomSheetType] = useState()
+
+  const dimensions = useWindowDimensions()
+
+  const bottomSheetRef = useRef(null)
+
+  const snapPoints = useMemo(() => ['95%'], [])
```
Comment (loc-nguyenthien) [inline template/src/components/DebugMenu.js:83]: I think we can move this to `constants` too

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: template/src/components/DebugMenu.js [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+  const codePush = useMemo(() => CODEPUSH_KEYS.find(item => item.dev === codePushKey), [codePushKey])
+
+  return (
+    <>
+      <Draggable
+        isCircle
+        renderColor={colors.primary}
+        renderSize={debugMenuSize}
+        animatedViewProps={{height: dimensions.height}}
```
Comment (TranQuangPhi) [inline template/src/components/DebugMenu.js:141]: don't use style inline

File: template/src/components/DebugMenu.js (hunk 4 above)
Comment (ThinhKimVo) [reply template/src/components/DebugMenu.js:33]: why we need to move to constants
these are not a magic number

File: template/src/components/DebugMenu.js (hunk 6 above)
Comment (ThinhKimVo) [reply template/src/components/DebugMenu.js:83]: don't need

File: template/src/components/DebugMenu.js (hunk 3 above)
Comment (ThinhKimVo) [reply template/src/components/DebugMenu.js:28]: only this file is used, so dont need to move to constant
const DEBUGMENU_SIZE = 50

File: template/src/components/DebugMenu.js (hunk 7 above)
Comment (ThinhKimVo) [reply template/src/components/DebugMenu.js:141]: it should be flexible,
eslint isn't warning style inline

File: template/src/components/DebugMenu.js (hunk 3 above)
Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js]: Okay

File: template/src/components/DebugMenu.js (hunk 4 above)
Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js:33]: But they will not change though all app life, so I think we can just move it to `constants` so we can use it somewhere else (if needed) without having to calls useMemo for this again

File: template/src/components/DebugMenu.js (hunk 6 above)
Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js:83]: So I think maybe we can move it below `debugMenuSize`, so we don't have to use `useMemo` for this

File: template/src/components/DebugMenu.js (hunk 4 above)
Comment (ThinhKimVo) [reply template/src/components/DebugMenu.js:33]: where using, can import directly {getApplicationName, getBuildNumber, getDeviceId} from 'device-info'.
useMemo for render once, it's clear and not effect to the performance.

Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js:34]: Then we can skip this for now

File: template/App.js (hunk 2 above)
Comment (loc-nguyenthien) [reply template/App.js]: Outdated

File: template/src/components/DebugMenu.js (hunk 5 above)
Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js]: Done

File: template/src/components/DebugMenu.js (hunk 6 above)
Comment (loc-nguyenthien) [reply template/src/components/DebugMenu.js]: Done

**Review by honghoangsts — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Pull request merged**